### PR TITLE
Feature/arrays

### PR DESCRIPTION
**Pull request by Dashkent**

### CODE REVIEW

File: staff/daria-potemkina/arrays/array-functions/includes.js [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+            return true;
+        }
+    }
+}else{
+    for(var i = index; i < array.length; i++){
+        if(array[i] === element){
+            return true;
+        }
+    }
+}
```
Comment (lichab) [inline staff/daria-potemkina/arrays/array-functions/includes.js:16]: hey Daria, 2 things here. First make sure to properly indent the code if not when it gets bigger it becomes really hard to read and understand :wink: 
Also you don't really need an if else statement here.  
```js 
if (!index) index = 0; 
//for loop here just one time
```

File: staff/daria-potemkina/arrays/array-functions/index-of.js [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+        }
+    }else {
+        for (var j = index; j < array.length; j++) {
+            if (array[j] === element) {
+                return j
+            }
+        }
+    }
+    return -1
+
```
Comment (lichab) [inline staff/daria-potemkina/arrays/array-functions/index-of.js:18]: same here

File: staff/daria-potemkina/arrays/array-functions/join.js [hunk 3]
```diff
@@ -0,0 +1,13 @@
+'use stict'
```
Comment (lichab) [inline staff/daria-potemkina/arrays/array-functions/join.js:1]: typo here  :laughing:  `use stict`

File: staff/daria-potemkina/arrays/array-functions/join.spec.js [hunk 4]
```diff
@@ -0,0 +1,47 @@
+describe('join', function () {
+
+    it('it should return a tring with elements of array separating for ,', function () {
```
Comment (lichab) [inline staff/daria-potemkina/arrays/array-functions/join.spec.js:3]: this should be de default behaviour when no separator is passed.

File: staff/daria-potemkina/arrays/array-functions/pop.js [hunk 5]
```diff
@@ -0,0 +1,13 @@
+'use strict';
+
+function pop(array) {
+    if (array.length > 0) {
+        var deleted = [array[array.length-1]];
```
Comment (lichab) [inline staff/daria-potemkina/arrays/array-functions/pop.js:5]: pop does not return an array with the deleted element, just the element itself

File: staff/daria-potemkina/arrays/array-functions/push.js [hunk 6]
```diff
@@ -0,0 +1,9 @@
+'use strict';
+
+function push(array, ...args){
```
Comment (lichab) [inline staff/daria-potemkina/arrays/array-functions/push.js:3]: the spread operator ` ...args` is an ES6 feature change it to be ES5

File: staff/daria-potemkina/arrays/array-functions/reduce.js [hunk 7]
```diff
@@ -0,0 +1,17 @@
...
+    var result = 0;
+    if (array.length !== 0) {
+        if (typeof initialValue === 'undefined') {
+            initialValue = 0;
+        }
+        for (var i = 0; i < array.length; i++) {
+            var acumulator = result;
+            result = expression(acumulator, array[i]);
+        }
+        result = initialValue + result;
```
Comment (lichab) [inline staff/daria-potemkina/arrays/array-functions/reduce.js:13]: you don't need this ` result = initialValue + result ` here, that would be part of the reducer function that the user passes if he wanted to do that.

File: staff/daria-potemkina/arrays/array-functions/reduce.js [hunk 8]
```diff
@@ -0,0 +1,17 @@
+'use strict'
+
+function reduce(array, expression, initialValue) {
+    var result = 0;
+    if (array.length !== 0) {
+        if (typeof initialValue === 'undefined') {
+            initialValue = 0;
+        }
+        for (var i = 0; i < array.length; i++) {
+            var acumulator = result;
```
Comment (lichab) [inline staff/daria-potemkina/arrays/array-functions/reduce.js:10]: if you take the accumulator out of the loop you don't need to declare result AND accumulator, just accumulator

File: staff/daria-potemkina/arrays/array-functions/reduce.js [hunk 9]
```diff
@@ -0,0 +1,17 @@
...
+            result = expression(acumulator, array[i]);
+        }
+        result = initialValue + result;
+        
+    }
+    return result;
```
Comment (lichab) [inline staff/daria-potemkina/arrays/array-functions/reduce.js:16]: reduce does not return 0 if the array length is 0

File: staff/daria-potemkina/arrays/array-functions/splice.js [hunk 10]
```diff
@@ -0,0 +1,50 @@
+'use strict'
+
+function splice(array, start, deleteCount, ...args) {
```
Comment (lichab) [inline staff/daria-potemkina/arrays/array-functions/splice.js:3]: again here, `...args` is a ES6 feature

File: staff/daria-potemkina/arrays/array-functions/includes.js (hunk 1 above)
Comment (Dashkent) [reply staff/daria-potemkina/arrays/array-functions/includes.js]: Lisandro, thank you for the feedback. Modified the code to make it more readable.

File: staff/daria-potemkina/arrays/array-functions/index-of.js (hunk 2 above)
Comment (Dashkent) [reply staff/daria-potemkina/arrays/array-functions/index-of.js]: changed

File: staff/daria-potemkina/arrays/array-functions/join.spec.js (hunk 4 above)
Comment (Dashkent) [reply staff/daria-potemkina/arrays/array-functions/join.spec.js:3]: added the defauld behaviour.

File: staff/daria-potemkina/arrays/array-functions/pop.js (hunk 5 above)
Comment (Dashkent) [reply staff/daria-potemkina/arrays/array-functions/pop.js]: changed

**Comment by lichab** [comment]
Good Job Daria!

**Pull request merged**